includes/Discovery.php

```diff
@@ -0,0 +1,308 @@
+<?php
+/**
+ * Class Discovery.
+ *
+ * Responsible for improved discovery of stories on the web.
+ *
+ * @package   Google\Web_Stories
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://github.com/google/web-stories-wp
+ */
+
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Web_Stories;
+
+use Google\Web_Stories\REST_API\Stories_Controller;
+use WP_Post;
+
+/**
+ * Discovery class.
+ */
+class Discovery {
+	/**
+	 * Initialize discovery functionality.
+	 *
+	 * @return void
+	 */
+	public function init() {
+		add_action(
+			'web_stories_story_head',
+			static function () {
```

Comment:
Why is this a static function again?

Reply:
[Docs](https://www.php.net/manual/en/functions.anonymous.php):

> As of PHP 5.4, anonymous functions may be declared statically. This prevents them from having the current class automatically bound to them. Objects may also not be bound to them at runtime.

Static anonymous functions (closures) have a slight performance benefit over non-static ones. So if we don't need access to `$this` within the function, it is a good practice to declare it as static.

Reply:
Why is it static at all? Below there are methods accessed on the class. We follow this pattern.

Reply:
There should also be test coverage for this.